### PR TITLE
PHP 8.1 support

### PR DESCRIPTION
**Pull request by boesing**
|    Q          |   A
|-------------- | ------
| BC Break      | no (only on `@internal` marked classes)
| New Feature   | yes

### Description

Initial support for PHP 8.1.

The following changes were additionally made:

- Support for `symfony/console` v4 was dropped
- Compatibility classes (which were marked as `@internal`) for `symfony/console` v4 were dropped
- Native types for classes (which are marked as `@internal`) were added to match symfony interface types
- Some more "strict" inline type-hints were removed to match symfony typehints (were initially added with psalm support but were more strict than symfonys types)

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -19,34 +19,25 @@
...
-        }
-    },
     "require": {
-        "php": "^7.3 || ~8.0.0",
-        "composer/package-versions-deprecated": "^1.10.99",
-        "psr/container": "^1.0",
-        "symfony/console": "^4.3 || ^5.1.2",
-        "symfony/event-dispatcher": "^4.0 || ^5.0",
+        "php": "^7.3 || ~8.0.0 || ~8.1.0",
+        "composer/package-versions-deprecated": "1.11.99.4",
```
Comment (Ocramius) [inline composer.json]: Should probably not be an exact version, but at least `^`

File: src/Input/ParamAwareInput.php [hunk 2]
```diff
@@ -39,31 +37,31 @@ public function getParameterOption($values, $default = false, bool $onlyParams =
     }
 
     /**
-     * @return null|string|string[]
+     * @return mixed
```
Comment (Ocramius) [inline src/Input/ParamAwareInput.php:40]: Do these really return `mixed`? :thinking:

File: src/Input/StandardQuestionTrait.php [hunk 3]
```diff
@@ -35,6 +35,10 @@
  * Consumers composing this trait can use this method to generate the initial
  * Question instance, and then further configure it (e.g., to add a normalizer
  * or validator).
+ *
+ * @see InputParamInterface
+ *
+ * @psalm-require-implements InputParamInterface
```
Comment (Ocramius) [inline src/Input/StandardQuestionTrait.php:41]: Fancy: didn't know this one :O

**Review by Ocramius — CHANGES_REQUESTED** [review]

File: src/Input/ParamAwareInput.php (hunk 2 above)
Comment (boesing) [reply src/Input/ParamAwareInput.php:40]: Sadly, it does.
https://github.com/symfony/console/blob/8b1008344647462ae6ec57559da166c2bfa5e16a/Input/InputInterface.php#L58

Comment (Ocramius) [reply src/Input/ParamAwareInput.php:40]: 😢

**Review by Ocramius — APPROVED** [review]

**Comment by Ocramius** [comment]
Thanks @boesing!

**Pull request merged**